Code for Interpolation Search algorithm in C++ and Python Added

sharur7 opened the pull request:
<!--
Thank you for sending the PR! We appreciate you spending the time to work on these changes.

Help us understand your motivation by explaining why you decided to make this change.
You can learn more about contributing to NeoAlgo here: https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md

Happy Contributing!

-->

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

The Interpolation Search Algorithm:
Interpolation search is an algorithm for searching for a key in an array that has been ordered by numerical values assigned to the keys.
It is an improved variant of binary search.
It has not yet been added to this repository.
I have added code for this in C++ and Python.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#451

sharur7 committed:
Interpolation Search Code In C++ And Python Added

sharur7 committed:
Added Interpolation Search Algo Code in C++ & Python

sharur7 committed:
Updated README.md

sharur7 committed:
Updated Interpolation_Search.cpp

sharur7 commented:
@plazzy99 @Kajol-Kumari @Abhijit2505 Please review this PR. Thank You!

iamrajiv reviewed (changes requested):
@sharur7 please format the code properly since there is a lot of indentations try to use code formatters. Also, avoid using `lo` and `hi` rather try to use a meaningful variable like `low` and `high`. Add sample input-output and time and space complexity at the end of the code.

sharur7 committed:
Updated Interpolation_Search.cpp

a) Changed "lo" to "low" and "hi" to "high".
b) Beautified the code.
c) Added sample input and output code.
d) Added time & space complexity at the end of the code.

iamrajiv commented on C-Plus-Plus/search/Interpolation_Search.cpp:
```diff
@@ -1,78 +1,102 @@
+}
+
+
+//SAMPLE INPUT:
```
@sharur7 update this to multi-line comment something like this
```
/*
...
*/
```

sharur7 replied:
Yes, I did the changes in both C++ and Python files.

sharur7 committed:
Updated Interpolation_Search.py

a) Changed "lo" to "low" and "hi" to "high".
b) Beautified the code.
c) Added sample input and output code.
d) Added time & space complexity at the end of the code.

sharur7 committed:
Updated Interpolation_Search.cpp

sharur7 committed:
Updated Interpolation_Search.py

Added multi-line comment.

siddharth25pandey reviewed (approved):
@sharur7 Looks fine to me.
Able to merge.
Thanks!

SKAUL05 commented on Python/search/Interpolation_Search.py:
```diff
@@ -0,0 +1,103 @@
+# If the searching element is present in array, then returns index of it, else returns -1
+
+
+def interpolationSearch(arr, n, x):
```
Use proper naming convention as per PEP8

sharur7 replied:
Done

SKAUL05 commented on Python/search/Interpolation_Search.py:
```diff
@@ -0,0 +1,103 @@
+# Entering the elements in the array
+
+print 'Enter elements :'
+for i in range(0, n):
```
Use meaningful variables instead of alphabets

sharur7 replied:
Done

SKAUL05 commented on Python/search/Interpolation_Search.py:
```diff
@@ -0,0 +1,103 @@
+# Creating array of the size n
+# Size would be taken into the variable n from user
+
+arr = []
```
Include this whole part under __name__ == "__main__"

sharur7 replied:
I defined main() function to the specified code.
Please check the commit "Updated Interpolation_Search.cpp". I made a mistake in commit the message it should end with .py extension.

sharur7 committed:
Updated Interpolation_Search.py

a) Meaningful variable name
b) Used PEP8 conventions

sharur7 committed:
Updated Interpolation_Search.cpp

Included specified code into main() function

SKAUL05 commented on Python/search/Interpolation_Search.py:
```diff
@@ -0,0 +1,96 @@
+    # Entering the elements in the array
+
+    print ('Enter elements :')
+    for i in range(0, size):
```
```suggestion
    for index in range(0, size):
```

sharur7 replied:
Done

sharur7 committed:
Updated Interpolation_Search.py

Changed 'i' to 'index'

iamrajiv commented:
@all-contributors please add @sharur7 for code
